<?xml version="1.0" ?>

<launch>
    <arg name="robot_type" default="a1" doc="Robot type: [a1, aliengo, go1, laikago]"/>
    <arg name="cheater" default="false"/>

    <!-- make the files into global parameters -->
    <param name="urdfFile" value="/tmp/legged_control/$(arg robot_type).urdf"/>
    <param name="taskFile" value="$(find legged_controllers)/config/$(arg robot_type)/task.info"/>
    <param name="referenceFile" value="$(find legged_controllers)/config/$(arg robot_type)/reference.info"/>
    <param name="gaitCommandFile" value="$(find legged_controllers)/config/$(arg robot_type)/gait.info"/>

    <rosparam file="$(find legged_controllers)/config/controllers.yaml" command="load"/>

    <node if="$(arg cheater)" name="controller_loader" pkg="controller_manager" type="controller_manager"
          output="screen" args="load
          controllers/joint_state_controller
          controllers/legged_controller
          controllers/legged_cheater_controller
"/>

    <node unless="$(arg cheater)" name="controller_loader" pkg="controller_manager" type="controller_manager"
          output="screen" args="spawn
          controllers/joint_state_controller
          controllers/legged_controller
"/>

    <node pkg="legged_controllers" type="legged_gait_publisher" name="legged_robot_gait"
          output="screen"/>

    <node pkg="legged_controllers" type="legged_target_trajectories_publisher" name="legged_robot_target_trajectories"
          output="screen"/>

    <node pkg="joy" type="joy_node" name="joy_node">
        <param name="deadzone" value="2e-1"/>
        <param name="autorepeat_rate" value="1000"/>
        <param name="default_trig_val" value="true"/>
    </node>

    <node pkg="legged_controllers" type="legged_joystick_command_publisher" name="legged_joystick_command_publisher">
        <param name="linear_vel_scale" value="1.0"/>
        <param name="angular_vel_scale" value="1.0"/>
    </node>
</launch>
